File: 프로그래머스/3/12979. 기지국 설치/기지국 설치.py
def solution(n, stations, w):
    answer = 0
    idx, location = 0, 1
    while location <= n:
        if idx < len(stations) and location >= stations[idx] - w:
            location = stations[idx] + w + 1
            idx += 1
        else:
            location += 2 * w + 1
            answer += 1
    return answer

File: 프로그래머스/3/12979. 기지국 설치/기지국 설치.java
class Solution {
    public int solution(int n, int[] stations, int w) {
        int idx, location, answer;
        idx = 0; location = 1; answer = 0;
        while (location <= n) {
            if (idx < stations.length && location >= stations[idx] - w) {
                location = stations[idx] + w + 1;
                idx++;
            } else {
                location += 2 * w + 1;
                answer++;
            }
        }
        return answer;
    }
}